pkg/workflow/controllers/deadline_reconciler.go

```diff
@@ -55,6 +55,11 @@ func (it *DeadlineReconciler) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, client.IgnoreNotFound(err)
 	}
 
+	if ConditionEqualsTo(node.Status, v1alpha1.ConditionDeadlineExceed, corev1.ConditionTrue) {
```

Comment:
This sync can ensure that the condition of the child node is consistent with the parent node, but I still don't understand when the child will be overwritten, and this behavior is not what we expected?

Reply:
chaos_node_reconciler depends on `WorkflowFinished()` to decide the Chaos CR should exist or not, and `WorkflowFinished()` make the decision based on condition `ConditionAccomplished` and `ConditionDeadlineExceed`. So here is the simplified routine for reproducing this issue:

1. there are two WorkflowNode: one is Serial, the other one is PodChaos controlled by the Serial one. The Serial one has `deadline: 5s`, and the PodChaos one has `dedline:10s`. 
2. after 5 seconds, the Serial one trigs the deadline exceeds checks, and propagate this condition to the PodChaos one, the condition `ConditionDeadlineExceed` of PodChaos is set to `true` with reason `ParentNodeDeadlineExceed`.
3. ChaosNodeReconciler works on the Serial one, `WorkflowFinished()` return `true`, the related Chaos CR is deleted.
4. (unexpected behavior) DeadlineReconciler reconcile the Serial one again, but the current timestamp is earlier than itself's deadline, `ConditionDeadlinExceed` was overwrite from `true, reason: ParentNodeDeadlineExceed` to `false, reason: NodeDeadlineNotExceed`
5. (unexpected behavior) ChaosNodeReconciler works on this Serial one, `WorkflowFinished()` return `false`, the related Chaos CR is created again.

Reply:
> but the current timestamp is earlier than itself's deadline 

If the node is the origin PodChaos node, the current timestamp should be after `deadline`. This node is recreated?

Reply:
> > but the current timestamp is earlier than itself's deadline
> 
> If the node is the origin PodChaos node, the current timestamp should be after `deadline`. This node is recreated?

Yes. But it would not be reproduced stably. Because the reconciliation of different reconcile happens parallelly, the step 3 4 5 is the only way to reproduce this issue.